velox/dwio/common/BufferedInput.h

```diff
@@ -17,30 +17,33 @@
 #pragma once
 
 #include "velox/dwio/common/DataBuffer.h"
+#include "velox/dwio/common/IoStatistics.h"
+#include "velox/dwio/common/Options.h"
 #include "velox/dwio/common/SeekableInputStream.h"
 #include "velox/dwio/common/StreamIdentifier.h"
 
 namespace facebook::velox::dwio::common {
 
 class BufferedInput {
  public:
-  constexpr static uint64_t kMaxMergeDistance = 1024 * 1024 * 1.25;
-
   BufferedInput(
       std::shared_ptr<ReadFile> readFile,
       memory::MemoryPool& pool,
       const MetricsLogPtr& metricsLog = MetricsLog::voidLog(),
-      IoStatistics* FOLLY_NULLABLE stats = nullptr)
+      IoStatistics* FOLLY_NULLABLE stats = nullptr,
+      int32_t mergeDistance = ReaderOptions::kMaxMergeDistance)
       : input_{std::make_shared<ReadFileInputStream>(
             std::move(readFile),
             metricsLog,
             stats)},
-        pool_{pool} {}
+        pool_{pool},
+        mergeDistance_(mergeDistance) {}
 
   BufferedInput(
       std::shared_ptr<ReadFileInputStream> input,
-      memory::MemoryPool& pool)
-      : input_(std::move(input)), pool_(pool) {}
+      memory::MemoryPool& pool,
+      int32_t mergeDistance = ReaderOptions::kMaxMergeDistance)
```

Comment:
cleaner to pass `ReaderOptions` in this API.

Reply:
thanks.

Reply:
Can we move the refactor of `maxMergeDistance` to ReaderOptions to its own PR since it is independent?

Reply:
> Can we move the refactor of maxMergeDistance to ReaderOptions to its own PR since it is independent?

Sure, I will revert and open another PR, thanks!